Lib/typing.py

```diff
@@ -3265,3 +3266,82 @@ def reveal_type(obj: T, /) -> T:
     """
     print(f"Runtime type is {type(obj).__name__!r}", file=sys.stderr)
     return obj
+
+
+def dataclass_transform(
+    *,
+    eq_default: bool = True,
+    order_default: bool = False,
+    kw_only_default: bool = False,
+    field_specifiers: tuple[type[Any] | Callable[..., Any], ...] = (),
+    **kwargs: Any,
+) -> Callable[[T], T]:
+    """Decorator that marks a function, class, or metaclass as providing
+    dataclass-like behavior.
+
+    Example:
+
+        from typing import dataclass_transform
+
+        _T = TypeVar("_T")
+
+        # Used on a decorator function
```

Comment:
The comments don't demarcate the sections clearly. Doing how it's like in the PEP feels clearer to me. I suggest  something like this instead:
```
Examples:
# NOTE: I removed the dataclass_transform and _T = TypeVar("_T") parts

Used on a decorator function:
    @dataclass_transform()
...
Used on a base class:
    @dataclass_transform()
...
Used on a metaclass:
    @dataclass_transform()
...
```

Reply:
Good point. I duplicated the TypeVar declaration in each example to keep it clear what `_T` refers to.